store/tikv/kv.go

```diff
@@ -347,3 +395,60 @@ func (s *KVStore) SetTiKVClient(client Client) {
 func (s *KVStore) GetTiKVClient() (client Client) {
 	return s.client
 }
+
+func (s *KVStore) getMinResolveTSByStores(stores []*Store) uint64 {
+	failpoint.Inject("injectResolveTS", func(val failpoint.Value) {
+		injectTS := val.(int)
+		failpoint.Return(uint64(injectTS))
+	})
+	minSafeTS := uint64(math.MaxUint64)
+	s.resolveTSMu.RLock()
+	defer s.resolveTSMu.RUnlock()
+	for _, store := range stores {
+		safeTS := s.resolveTSMu.resolveTS[store.storeID]
+		if safeTS < minSafeTS {
+			minSafeTS = safeTS
+		}
+	}
+	return minSafeTS
```

Comment:
What if:
1. A new store is just added into the zone.
2. The store just begins to replicate data to itself.
3. The minSafeTS is very stale, and TiDB has to read with minStartTS.
4. When TiDB requests data from the new store with minStartTS, the store will always fail until it catches up with the leader.

Reply:
Oh, so the replica is a learner, and TiDB won't read it.

Reply:
If the resolveTS is too stale, we will use minStartTS directly.